pkg/webui/console/components/pubsub-form/validation-schema.js

```diff
@@ -71,14 +72,14 @@ export default Yup.object().shape({
         .matches(mqttUrlRegexp, sharedMessages.validateUrl)
         .required(sharedMessages.validateRequired),
       client_id: Yup.string()
-        .matches(idRegexp, sharedMessages.validateIdFormat)
+        .matches(noSpacesRegexp, sharedMessages.validateNoSpaces)
         .min(2, sharedMessages.validateTooShort)
         .max(23, sharedMessages.validateTooLong)
         .required(sharedMessages.validateRequired),
       username: Yup.string().when('use_credentials', {
         is: true,
         then: Yup.string()
-          .matches(idRegexp, sharedMessages.validateIdFormat)
+          .matches(noSpacesRegexp, sharedMessages.validateNoSpaces)
```

Comment:
For both `client_id` and `username` there are no such restrictions enforces by the backend. However, according to the spec:

> The ClientId MUST be a UTF-8 encoded string ...

> The Server MUST allow ClientIds which are between 1 and 23 UTF-8 encoded bytes in length, and that contain only the characters
> "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"

Note, no special characters.

cc @adriansmares, @htdvisser  I guess we should make the backend validation more strict to comply with the spec.

Reply:
Also from the spec:

> The Server MAY allow ClientID’s that contain more than 23 encoded bytes. The Server MAY allow ClientID’s that contain characters not included in the list given above.